client/js/app/stores/ExplorerStore.js

```diff
@@ -84,6 +101,99 @@ function _getDefaultFilterCoercionType(explorer, filter) {
 }
 
 /**
+ * Runs through the changeset and moves data around if necessary
+ * @param {Object} explorer   The explorer model that is being updated
+ * @param {Object} updates    The updated explorer model
+ * @return {Object}           The new set of updates
+ */
+function _prepareUpdates(explorer, updates) {
+  var newModel = _.assign({}, explorer, updates);
+
+  newModel = _removeEmailExtractionFields(explorer, newModel);
+  newModel = _migrateToFunnel(explorer, newModel);
+  newModel = _migrateFromFunnel(explorer, newModel);
+
+  return newModel;
+}
+
+/**
+ * If the query got changed to a funnel, move the step-specific parameters to a steps object.
+ * @param {Object} explorer The explorer model that is being updated
+ * @param {Object} newModel The updated explorer model
+ * @return {Object}         The new set of updates
+ */
+function _migrateToFunnel(explorer, newModel) {
+  var sharedProperties = ['event_collection', 'time', 'timezone', 'filters']
+  if(newModel.query.analysis_type === 'funnel' && explorer.query.analysis_type !== 'funnel') {
+    // Changing TO funnels
+    var firstStep = _defaultStep();
+    firstStep.active = true;
+
+    _.each(sharedProperties, function (key) {
+      if(!_.isUndefined(explorer.query[key]) && !_.isNull(explorer.query[key])) {
+        firstStep[key] = explorer.query[key] 
+      }      
+
+      delete newModel.query[key]
+    });
+
+    if(!_.isUndefined(explorer.query.target_property) && !_.isNull(explorer.query.target_property)) {
+      firstStep.actor_property = explorer.query.target_property;
+      delete explorer.query.target_property;
+    }
+
+    newModel.query.steps = [firstStep];
+  }
+
+  return newModel;
+}
+
+/**
+ * If the query got changed from a funnel, move the applicable parameters out to the root query
+ * @param {Object} explorer The explorer model that is being updated
+ * @param {Object} newModel The updated explorer model
+ * @return {Object}         The new set of updates
+ */
+function _migrateFromFunnel(explorer, newModel) {
+  var sharedProperties = ['event_collection', 'time', 'timezone', 'filters']
+  if (newModel.query.analysis_type !== 'funnel' && explorer.query.analysis_type === 'funnel') {
+    // Changing FROM funnels
+    var activeStep = _.find(explorer.query.steps, function (step) {
+      return step.active
+    });
+
+    _.each(sharedProperties, function (key) {
+      if(!_.isUndefined(activeStep[key])) {
```

Comment:
The lodash `isUndefined` function is pretty straightforward:
```
function isUndefined(value) {
      return value === undefined;
    }
```

So for these if statements I think maybe a simple `if(activeStep[key])` or `if(activeStep[key] !== undefined)` should be ok. As we move closer towards hopefully using Babel+ES6 plus, it would be great to drop the need for these utility libs.